### PR TITLE
Add durable isolated orchestration item template

### PR DESCRIPTION
**Pull request by jviau**
Adds item template for durable functions orchestration in dotnet isolated. The template is a direct copy of the existing template for in-proc, with changes to make it compatible with isolated (using and type changes).

### CODE REVIEW

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:14]: I like that you added the logger as part of this template!

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:8]: This might be a dumb question, but are there any template parameters we need to consider and use for the class name? Or is it okay that we hardcode it and let developers rename it after its created?

**Review by cgillum — COMMENTED** [review]
I'm not really familiar with how item templates work so I can't really comment on the JSON content, but everything else pretty much looks good to me. I did have one question below, though:

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs
Comment (jviau) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:8]: That is a great question haha. I think we could definitely improve this template to parameterize the name. However, given the time constraints I would prefer to leave that for a later time.

**Review by cgillum — APPROVED** [review]
Signing off with the caveat that I'm not at all an expert on these templates. We should probably get an ACK from @soninaren before merging.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs
Comment (soninaren) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:9]: The string `DurableFunctionsOrchestrationCSharp` is actually value of the `source` property. VS actually prompts for the functionName and replaces the source property with user input.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:41]: I see this package is not yet published. Do we have planned date for publishing this?

Comment (jviau) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:41]: Yes - either later today or tomorrow.

**Review by soninaren — APPROVED** [review]

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json
Comment (jviau) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:41]: They are published now https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Extensions.DurableTask/1.0.0

**Pull request merged**
